### ISSUE TITLE
laf_is_inside_image uses w/h as the image extent while the rest of laf.py uses w-1/h-1

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4064]
### Describe the bug

`laf_is_inside_image` treats the image extent as `w` × `h`, while every other function in `kornia/feature/laf.py` treats it as `(w - 1)` × `(h - 1)`. LAFs that reach one pixel past the last valid coordinate are reported as fully inside.

https://github.com/kornia/kornia/blob/main/kornia/feature/laf.py#L527

```python
good_lafs_mask = (
    (pts[..., 0] >= border) * (pts[..., 0] <= w - border) * (pts[..., 1] >= border) * (pts[..., 1] <= h - border)
)
```

Every other extent computation in the same file uses `w - 1` / `h - 1`:

| line | function | extent |
|---|---|---|
| 339–340 | `denormalize_laf` | `w - 1`, `h - 1` |
| 370–371 | `normalize_laf` | `w - 1`, `h - 1` |
| 400–401 | `generate_patch_grid_from_normalized_LAF` | `w - 1`, `h - 1` |
| **527** | **`laf_is_inside_image`** | **`w`, `h`** |

This matches the documented convention in `get_laf_center`: *"the center of a 5-pixel image (coordinates from 0 to 4) is 2, and not 2.5"* — coordinates run `0 .. w-1`.

### Reproduction

```python
import torch
from kornia.feature.laf import laf_is_inside_image, laf_from_center_scale_ori

img = torch.zeros(1, 1, 32, 32)          # valid x coordinates are 0 .. 31
for cx in [29.0, 29.5, 30.0, 33.0]:
    xy = torch.tensor([[[cx, 16.0]]], dtype=torch.float64)
    laf = laf_from_center_scale_ori(xy, torch.full((1, 1, 1, 1), 2.0, dtype=torch.float64))
    print(f"centre x={cx:5.1f}, radius 2 -> rightmost boundary x={cx + 2:5.1f}  "
          f"inside={bool(laf_is_inside_image(laf, img)[0, 0])}")
```

```
centre x= 29.0, radius 2 -> rightmost boundary x= 31.0  inside=True
centre x= 29.5, radius 2 -> rightmost boundary x= 31.5  inside=True     <-- past 31
centre x= 30.0, radius 2 -> rightmost boundary x= 32.0  inside=True     <-- past 31
centre x= 33.0, radius 2 -> rightmost boundary x= 35.0  inside=False
```

### Expected behavior

The upper bound should be `w - 1 - border` / `h - 1 - border`, making the check symmetric with the `>= border` lower bound and consistent with the rest of `laf.py`.

### Additional context

The existing test does not discriminate the boundary — its "inside" LAF is a radius-1 circle at `(5, 2)` in a 10×5 image, well clear of the edge:

https://github.com/kornia/kornia/blob/main/tests/feature/test_laf.py#L521-L526

Fixing this drops LAFs within one pixel of the right/bottom border that are currently kept, so it changes detector output near image edges and wants a deliberate decision.

Found during an audit of `kornia.feature`.

Posted on behalf of @ducha-aiki by Claude (Opus 5).
